build-pex.sh

```diff
@@ -20,6 +20,6 @@ poetry export -f requirements.txt --without-hashes --output requirements.txt
 
 echo "Requirements exported..."
 
-pex . --requirement=requirements.txt -o dist/portable_cli -e moosecli
+pex . --requirement=requirements.txt --python-shebang="#!/usr/bin/env python3" -o dist/portable_cli -e moosecli
```

Comment:
@b-long choosing a shebang that works on many machines is hard, consider using `--sh-boot` instead if you can be more sure `/bin/sh` will be on all machines you deploy to. With modern Pex you can also avoid the problem altogether by specifying `--scie eager` instead to get a PEX that includes Python itself and is a single file native executable: https://docs.pex-tool.org/scie.html

Reply:
Hi @jsirois 👋  Thank you for the tip! 🥇 

I appreciate the information a lot, and I'm going to dig in and play with this `--scie eager` feature.  

A few quick (rabbit-hole) thoughts:
1. I think your PR / comment here is a fantastic way to teach/learn.  Thanks for opening this PR, it'll absolutely help me in learning more about Pex 🙂 I hope to see other PRs like this in the future (for me or anyone) to assist in open learning.
2. FWIW, I think we (the Python community) have to solve the binary distribution problem.  As long as the licensing is a proper open source license, I don't care much who "_wins_" (Pex, PyOxidizer, [SPy](https://github.com/spylang/spy), etc.) 
3. This might be a good way forward for me to address an issue I had with PyOxidizer ( https://github.com/b-long/moosecli/issues/5 ).  IIRC, Hynek had a similar issue as well building binaries with PyOxidizer ( `https://github.com/hynek/doc2dash/issues/138` )

Reply:
FYI PyOxidizer, if not dead, is dormant: https://gregoryszorc.com/blog/2024/03/17/my-shifting-open-source-priorities/

> 2. FWIW, I think we (the Python community) have to solve the binary distribution problem. 

I, personally, have given up on - not so much the Python community - but the PyPA in this regard. There is no apparent effort to push towards a solution, only efforts towards supporting marketplaces of solutions. I find this extremely maddening and I am a tool writer! I would love for my tool to die away and Python simply ship with the right tools - a singular set of them. The only truly thorny problem AFAICT is solved by PEP-517/518. Let the variation remain there in build systems, but unify everything else please. And don't count on a long drawn out 5, 10 year + thunderdome to hopefully settle winners via a marketplace. It's a supreme waste of time. Other communities simply don't have these self-inflicted issues.

> 3. This might be a good way forward for me to address an issue I had with PyOxidizer

Agreed. The science projects (https://github.com/a-scie) were born trying to solve the Python application bootstrapping problem for the Pants build tool, which initially went down the PyOxidizer route and ran into issues. Although the science solution has nothing to do with Python - it works for any language with a hermetic interpreter distribution - it was certainly motivated by shipping a Python app and the long dream of Pex being more like Google's internal PAR with an embedded interpreter.